bids/layout/index.py

```diff
@@ -31,12 +31,12 @@ def _check_path_matches_patterns(path, patterns):
     """Check if the path matches at least one of the provided patterns. """
     if not patterns:
         return False
-    path = os.path.abspath(path)
+    path = path.absolute()
     for patt in patterns:
-        if isinstance(patt, str):
+        if isinstance(patt, Path):
```

Comment:
Patterns are strings or regexes:

```suggestion
        if isinstance(patt, str):
```

Reply:
The `patterns` are *Path*s or regexes now:

- `index._check_path_matches_pattern` is called by `BIDSLayoutIndexer._validate_dir` and `BIDSLayoutIndexer._validate_file` only:
  - `patterns` are either `self._include_patterns` or `self._exclude_patterns` when called from `_validate_*` functions.
  - These are `None` by defaults and are popluated in `self.__call__` where they are set to whatever `validation.validate_indexing_args` returns.
  - That function only returns `Path` and `SRE_Pattern` objects so no pattern is ever of type str.